src/Cache/SourceAndOptionsHashCacheKeyGenerator.php

```diff
@@ -0,0 +1,15 @@
+<?php
+namespace ScriptFUSION\Porter\Cache;
+
+use ScriptFUSION\Porter\Options\EncapsulatedOptions;
+
+class SourceAndOptionsHashCacheKeyGenerator implements CacheKeyGenerator
+{
+    public function generateCacheKey($source, EncapsulatedOptions $options = null)
+    {
+        $optionsCopy = $options ? $options->copy() : [];
+        ksort($optionsCopy);
```

Comment:
The thought occurs that this is not a good interface because every implementation would need to call `$options->copy()` and every implementation would need to sort the options. Therefore the caller should copy and sort the options and pass an `array` as the second parameter.